test: failure in wallet_resendwallettransactions.py

https://cirrus-ci.com/task/5603470202896384?logs=ci#L2681

Current diff to reproduce fastest:

```diff
diff --git a/test/functional/wallet_resendwallettransactions.py b/test/functional/wallet_resendwallettransactions.py
index 26df0841d8..0b8f4bf317 100755
--- a/test/functional/wallet_resendwallettransactions.py
+++ b/test/functional/wallet_resendwallettransactions.py
@@ -86,6 +86,7 @@ class ResendWalletTransactionsTest(BitcoinTestFramework):
         child_txid = node.send(outputs=[{addr: 0.5}], options={"inputs": [{"txid":txid, "vout":0}]})["txid"]
         while True:
             txids = node.listreceivedbyaddress(minconf=0, address_filter=addr)[0]["txids"]
+            assert_equal(len(txids), 2)
             if txids == [child_txid, txid]:
                 break
             bumped = node.bumpfee(child_txid)
```

Log:

```
2022-09-14T14:15:56.480000Z TestFramework (INFO): Chain of unconfirmed not-in-mempool txs are rebroadcast
-29-> send {"outputs": [{"bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0": 0.5}], "options": {"inputs": [{"txid": "8d6c948eec2b1fbe8362df36cefd5ca803e5e42ebf7f52538ef26331b831ffeb", "vout": 0}]}}
<-29- [0.000320] {"txid": "97d516151b92011f82a612bfea0b70149c617d5ea516196e76c1f599afc1a785", "complete": true}
-30-> listreceivedbyaddress {"minconf": 0, "address_filter": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0"}
<-30- [0.000386] [{"address": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0", "amount": "1.50000000", "confirmations": 0, "label": "", "txids": ["8d6c948eec2b1fbe8362df36cefd5ca803e5e42ebf7f52538ef26331b831ffeb", "97d516151b92011f82a612bfea0b70149c617d5ea516196e76c1f599afc1a785"]}]
-31-> bumpfee ["97d516151b92011f82a612bfea0b70149c617d5ea516196e76c1f599afc1a785"]
<-31- [0.000279] {"txid": "ca38cf9d48be32f7576155d796192ba635b46daf6ac077879ad6d21296d357b8", "origfee": "0.00002820", "fee": "0.00003526", "errors": []}
-32-> removeprunedfunds ["97d516151b92011f82a612bfea0b70149c617d5ea516196e76c1f599afc1a785"]
<-32- [0.000256] null
-33-> listreceivedbyaddress {"minconf": 0, "address_filter": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0"}
<-33- [0.000236] [{"address": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0", "amount": "1.50000000", "confirmations": 0, "label": "", "txids": ["8d6c948eec2b1fbe8362df36cefd5ca803e5e42ebf7f52538ef26331b831ffeb", "ca38cf9d48be32f7576155d796192ba635b46daf6ac077879ad6d21296d357b8"]}]
-34-> bumpfee ["ca38cf9d48be32f7576155d796192ba635b46daf6ac077879ad6d21296d357b8"]
<-34- [0.000250] {"txid": "7436a7a7ccc63bb79e19ba212bb768de1261523fd4481343234430e1de5916af", "origfee": "0.00003526", "fee": "0.00004232", "errors": []}
-35-> removeprunedfunds ["ca38cf9d48be32f7576155d796192ba635b46daf6ac077879ad6d21296d357b8"]
<-35- [0.000240] null
-36-> listreceivedbyaddress {"minconf": 0, "address_filter": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0"}
<-36- [0.000229] [{"address": "bcrt1qv5q43yunpc7mjw73nje7nd0g66lpjts2xazdj0", "amount": "2.00000000", "confirmations": 0, "label": "", "txids": ["7436a7a7ccc63bb79e19ba212bb768de1261523fd4481343234430e1de5916af", "8d6c948eec2b1fbe8362df36cefd5ca803e5e42ebf7f52538ef26331b831ffeb", "97d516151b92011f82a612bfea0b70149c617d5ea516196e76c1f599afc1a785"]}]
2022-09-14T14:15:56.500000Z TestFramework (ERROR): Assertion failed
    assert_equal(len(txids), 2)
AssertionError: not(3 == 2)

--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -882,7 +882,7 @@ bool CWallet::MarkReplaced(const uint256& originalHash, const uint256& newHash)
 
     wtx.mapValue["replaced_by_txid"] = newHash.ToString();
 
-    // Refresh mempool status without waiting for transactionRemovedFromMempool
+    // Refresh mempool status without waiting for transactionRemovedFromMempool or transactionAddedToMempool
     RefreshMempoolStatus(wtx, chain());
 
     WalletBatch batch(GetDatabase());

--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -1920,7 +1920,7 @@ std::set<uint256> CWallet::GetTxConflicts(const CWalletTx& wtx) const
 // The `force` option results in all unconfirmed transactions being submitted to
 // the mempool. This does not necessarily result in those transactions being relayed,
 // that depends on the `relay` option. Periodic rebroadcast uses the pattern
-// relay=true force=false (also the default values), while loading into the mempool
+// relay=true force=false, while loading into the mempool
 // (on start, or after import) uses relay=false force=true.
 void CWallet::ResubmitWalletTransactions(bool relay, bool force)
 {

--- a/test/functional/wallet_resendwallettransactions.py
+++ b/test/functional/wallet_resendwallettransactions.py
@@ -98,13 +98,13 @@ class ResendWalletTransactionsTest(BitcoinTestFramework):
         block = create_block(int(node.getbestblockhash(), 16), create_coinbase(node.getblockcount() + 1), block_time)
         block.solve()
         node.submitblock(block.serialize().hex())
+        # Set correct m_best_block_time, which is used in ResubmitWalletTransactions
         node.syncwithvalidationinterfacequeue()
 
         # Evict these txs from the mempool
         evict_time = block_time + 60 * 60 * DEFAULT_MEMPOOL_EXPIRY_HOURS + 5
         node.setmocktime(evict_time)
         indep_send = node.send(outputs=[{node.getnewaddress(): 1}], options={"inputs": [indep_utxo]})
-        node.syncwithvalidationinterfacequeue()
         node.getmempoolentry(indep_send["txid"])
         assert_raises_rpc_error(-5, "Transaction not in mempool", node.getmempoolentry, txid)
         assert_raises_rpc_error(-5, "Transaction not in mempool", node.getmempoolentry, child_txid)

--- a/test/functional/wallet_resendwallettransactions.py
+++ b/test/functional/wallet_resendwallettransactions.py
@@ -89,6 +89,10 @@ class ResendWalletTransactionsTest(BitcoinTestFramework):
             if txids == [child_txid, txid]:
                 break
             bumped = node.bumpfee(child_txid)
+            # The scheduler queue creates a copy of the added tx after
+            # send/bumpfee and re-adds it to the wallet (undoing the next
+            # removeprunedfunds). So empty the scheduler queue:
+            node.syncwithvalidationinterfacequeue()
             node.removeprunedfunds(child_txid)
             child_txid = bumped["txid"]
         entry_time = node.getmempoolentry(child_txid)["time"]

